### PR TITLE
Chat 18  validate login and registration

### PR DESCRIPTION
**Pull request by DmytroTeliukov**

### CODE REVIEW

File: src/main/java/com/chat/yourway/controller/AuthenticationController.java [hunk 1]
```diff
@@ -27,8 +28,8 @@ public class AuthenticationController {
       consumes = MediaType.APPLICATION_JSON_VALUE)
   @Operation(summary = "Registration")
   public AuthResponseDto register(@RequestBody ContactRequestDto request,
-      HttpServletRequest httpRequest) {
-    return authService.register(request, httpRequest);
+                                  @RequestHeader(HttpHeaders.REFERER) String clientAddress) {
```
Comment (Troha7) [inline src/main/java/com/chat/yourway/controller/AuthenticationController.java]: Could you rename clientAddress -> clientUrl for better readability

File: src/main/java/com/chat/yourway/dto/request/ContactRequestDto.java [hunk 2]
```diff
@@ -14,9 +17,11 @@
 @Builder
 @ToString
 public class ContactRequestDto {
-
+  @Username
   private String username;
+  @ContactEmail
   private String email;
+  @Password
   private String password;
```
Comment (Troha7) [inline src/main/java/com/chat/yourway/dto/request/ContactRequestDto.java]: Could you rename annotations for better readability (exemple: @Password -> @ValidatePassword)

File: src/test/java/com/chat/yourway/integration/extension/PostgresExtension.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
 public class PostgresExtension implements Extension, BeforeAllCallback, AfterAllCallback {
 
   private static final PostgreSQLContainer<?> container = new PostgreSQLContainer<>(
-      DockerImageName.parse("postgres:14.6"))
+      DockerImageName.parse("postgres:latest"))
```
Comment (Troha7) [inline src/test/java/com/chat/yourway/integration/extension/PostgresExtension.java:13]: It is better to use the same database version in the container as in the working database

**Comment by Troha7** [comment]
Can you add a validation exception handler, for handling validation exceptions? For example: (@ExceptionHandler(MethodArgumentNotValidException.class))
Read this link:
https://salithachathuranga94.medium.com/validation-and-exception-handling-in-spring-boot-51597b580ffd

File: src/test/java/com/chat/yourway/integration/extension/PostgresExtension.java (hunk 3 above)
Comment (DmytroTeliukov) [reply src/test/java/com/chat/yourway/integration/extension/PostgresExtension.java:13]: I tried to run integration tests with ver. 14.6 and I got errors. Then I changed to latest and tests were passed. As for me, latest version gives me more confidence and reliability.

**Comment by DmytroTeliukov** [comment]
Should I set MethodArgumentNotValidException instead of IllegalArgumentException in Validator classes?

**Review by Troha7 — APPROVED** [review]
Ok)

**Pull request merged**